packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -0,0 +1,236 @@
+import * as React from 'react';
+import { useId, useControllableValue, useMount } from '@fluentui/react-utilities';
+import {
+  getCode,
+  ArrowDownKey,
+  ArrowUpKey,
+  ArrowLeftKey,
+  ArrowRightKey,
+  PageDownKey,
+  PageUpKey,
+  HomeKey,
+  EndKey,
+} from '@fluentui/keyboard-key';
+import { SliderState } from './Slider.types';
+
+/**
+ * Validates that the `value` is a number and falls between the min and max.
+ *
+ * @param value - the value to be clamped
+ * @param min - the lowest valid value
+ * @param max - the highest valid value
+ */
+const clamp = (value: number, min: number, max: number): number => Math.max(min, Math.min(max, value || 0));
+
+/**
+ * Gets the current percent of specified value between a min and max
+ *
+ * @param value - the value to find the percent
+ * @param min - the lowest valid value
+ * @param max - the highest valid value
+ */
+const getPercent = (value: number, min: number, max: number) => {
+  return max === min ? 0 : ((value - min) / (max - min)) * 100;
+};
+
+const on = (element: Element, eventName: string, callback: (ev: Event) => void) => {
+  element.addEventListener(eventName, callback);
+  return () => element.removeEventListener(eventName, callback);
+};
+
+export const useSliderState = (state: SliderState): SliderState => {
+  const {
+    as = 'div',
+    value,
+    defaultValue = 0,
+    min = 0,
+    max = 10,
+    step = 1,
+    snapToStep = false,
+    ariaValueText,
+    onChange,
+    onPointerDown: onPointerDownCallback,
+    onKeyDown: onKeyDownCallback,
+  } = state;
+
+  const [currentValue, setCurrentValue] = useControllableValue(
+    value && clamp(value, min, max),
+    clamp(defaultValue, min, max),
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any -- ev is not a SyntheticEvent
+    (ev: any, val) => onChange?.(val!, ev),
+  );
+
+  const railRef = React.useRef<HTMLDivElement>(null);
+  const thumbRef = React.useRef<HTMLDivElement>(null);
+  const disposables = React.useRef<(() => void)[]>([]);
+  const id = useId('Slider', state.id);
```

Comment:
```suggestion
  const id = useId('slider-', state.id);
```

_nit_: In other converged components are we use kebab case for ids.